### PR TITLE
Don't create V6 binds if global V6 flag is off

### PR DESCRIPTION
**Pull request by bleggett**
Existing code handles the case where V6 is globally enabled but not enabled in the local pod context.

It does not handle the case where V6 is globally _disabled_ but is enabled in the local pod context, which is a tad surprising.

Only the DNS handler is affected by this.

### CODE REVIEW

File: src/dns/server.rs [hunk 1]
```diff
@@ -89,10 +89,21 @@ impl Server {
...
-            true
-        }));
+        // if the address we got from config is supposed to be v6-enabled,
+        // actually check if the local pod context our socketfactory operates in supports V6.
+        // This is to ensure globally-enabled V6 support doesn't try to create V6 addresses in pods
+        // that do not support it, and also ensure globally-disabled V6 support doesn't create V6 address
+        // even if that's turned off.
+        let local_address = if let config::Address::Localhost(v6_global_enable, _) = address {
+            if v6_global_enable {
+                address.with_ipv6(socket_factory.ipv6_enabled_localhost().unwrap_or_else(|e| {
```
Comment (howardjohn) [inline src/dns/server.rs:99]: nit: maybe we should put this in with_ipv6 and maybe call it something like "maybe disable ipv6"?

**Review by howardjohn — APPROVED** [review]
Nice find.

File: src/dns/server.rs (hunk 1 above)
Comment (bleggett) [reply src/dns/server.rs:99]: Yeah it's a bit too cute doing it out here, agreed.

**Pull request merged**